codepropertygraph/src/main/scala/io/shiftleft/passes/ParallelCpgPass.scala

```diff
@@ -48,6 +52,11 @@ abstract class ParallelCpgPass[T](cpg: Cpg, outName: String = "", keyPools: Opti
       writer.enqueue(None, None)
       writerThread.join()
     }
+
+    exceptionCaught.foreach { exception =>
+      baseLogger.warn(s"Exception in parallel CPG pass $name:", exception)
```

Comment:
The exception is already logged in line 50.

Reply:
No it's not... that's what I first thought as well, but because we start a new Thread (line 39), and the exception is thrown in that thread (and handled by the newly registered `UncaughtExceptionHandler`)